src/libraries/System.IO.FileSystem/tests/FileStream/CharacterDevice.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    [PlatformSpecific(TestPlatforms.AnyUnix)]
+    public class CharacterDevice
+    {
+        const string CharacterDevicePath = "/dev/tty";
```

Comment:
I thought the reason of getting ENXIO was because we were using a character device. I tried to repro with `/dev/null` and `/dev/zero` but I couldn't so I don't know what's special about this vs the other two character devices.

Reply:
It was the issue with `/dev/tty` (aka text/input environment) and `/dev/console` (aka physical terminal).
`/dev/null` has special handling https://man7.org/linux/man-pages/man4/null.4.html, I think that's why it bails early as a no-op.

Reply:
Maybe we also want a test using some special files against the `File.Write*` APIs that also use `RandomAccess` internally.